bundles/org.openhab.binding.souliss/src/main/java/org/openhab/binding/souliss/internal/handler/SoulissGatewayHandler.java

```diff
@@ -0,0 +1,324 @@
+            SoulissBindingSendDispatcherJob soulissSendDispatcherRunnable = new SoulissBindingSendDispatcherJob(bridge);
+            scheduler.scheduleWithFixedDelay(soulissSendDispatcherRunnable, 15,
+                    SoulissBindingConstants.SEND_DISPATCHER_MIN_DELAY_CYCLE_IN_MILLIS, TimeUnit.MILLISECONDS);
+        }
+    }
+
+    private int gwIpByte() {
+        return Integer.parseInt(ipAddressOnLAN.split("\\.")[3]);
+    }
+
+    @Override
+    public void handleRemoval() {
+        SoulissBindingNetworkParameters.removeGateway((byte) gwIpByte());
+        logger.debug("Gateway handler removing");
+    }
+
+    @Override
+    public void thingUpdated(Thing thing) {
+        logger.debug("Thing Updated: {}", thing.getThingTypeUID());
+        SoulissBindingNetworkParameters.removeGateway((byte) gwIpByte());
+        this.thing = thing;
+    }
+
+    public void dbStructAnswerReceived() {
+        soulissCommands.sendTypicalRequestFrame(SoulissBindingNetworkParameters.getDatagramSocket(), ipAddressOnLAN,
+                nodeIndex, userIndex, nodes);
+    }
+
+    public void setNodes(int nodes) {
+        this.nodes = nodes;
+    }
+
+    int iPosNodeSlot = 2;
+
+    public int getNodes() {
+        Thing thing;
+        int maxNode = 0;
+        Iterator<Thing> iterator = bridge.getThings().iterator();
+        while (iterator.hasNext()) {
+            thing = iterator.next();
+            String[] uuidStrings = thing.getUID().getAsString().split(SoulissBindingConstants.UUID_NODE_SLOT_SEPARATOR);
+            String[] uuidNodeNumber = uuidStrings[0].split(SoulissBindingConstants.UUID_ELEMENTS_SEPARATOR);
+
+            iPosNodeSlot = 2; // if uuid is of type souliss:gateway:[typical]:[node]-[slot] then node/slot is at
+                              // position 2
+            if (uuidNodeNumber.length > 3) {
+                iPosNodeSlot = 3;
+            }
+            if (Integer.parseInt(uuidNodeNumber[iPosNodeSlot]) > maxNode) {
+                maxNode = Integer.parseInt(uuidNodeNumber[iPosNodeSlot]);
+            }
+            // alla fine la lunghezza della lista sarà uguale al numero di nodi presenti
+        }
+        return maxNode + 1;
+    }
+
+    public void setMaxTypicalXnode(int maxTypicalXnode) {
+        this.maxTypicalXnode = maxTypicalXnode;
+    }
+
+    public int getMaxTypicalXnode() {
+        return maxTypicalXnode;
+    }
+
+    /**
+     * The {@link gatewayDetected} is used to notify that UDPServer decoded a Ping Response from gateway
+     *
+     * @author Tonino Fazio - Initial contribution
+     * @author Luca Calcaterra - Refactor for OH3
+     */
+
+    public void gatewayDetected() {
+        logger.debug("Setting Gateway ONLINE");
+        try {
+            updateStatus(ThingStatus.ONLINE);
+        } catch (Exception e) {
+            logger.debug("Illegal status transition to ONLINE");
+        }
```

Comment:
This could be removed as `updateStatus()` doesn't throw an exception.

Reply:
fixed